bakta/features/cds.py

```diff
@@ -473,4 +480,443 @@ def revise_special_cases_annotated(genome: dict, cdss: Sequence[dict]):
+                            tmp_cause.append('Translation exception: Selenocysteine around ' + ', '.join(map(str, cause['selenocysteine'])) + '.')
+                        if cause['pyrolysine']:  # only for pseudogenes with translation exception + other cause
+                            tmp_cause.append('Translation exception: Pyrolysin around ' + ', '.join(map(str, cause['pyrolysine'])) + '.')
+                        cds['pseudogene']['note'] = ' '.join(tmp_cause)  # pseudogene cause
+
+                        log.info(
+                            'Pseudogene: contig=%s, start=%i, stop=%i, strand=%s, casue=%s',
+                            cds['contig'], cds['start'], cds['stop'], cds['strand'], cds['pseudogene']['note']
+                        )
+
+                    elif cause['selenocysteine'] or cause['pyrolysine']:
+                        # TODO handle translation exception, correct annotation
+                        if cause['selenocysteine']:
+                            pass
+                        elif cause['pyrolysine']:
+                            pass
+                    else:  # skip non-extended genes (complete aa insertion or deletion)
+                        log.info(
+                            'No pseudogene: contig=%s, start=%i, stop=%i, strand=%s',
+                            cds['contig'], cds['start'], cds['stop'], cds['strand']
+                        )
+
+    pseudogenes: List[dict] = []
+    for cds in candidates:
+        if 'pseudogene' in cds:
+            pseudogenes.append(cds)
+            cds.pop('hypothetical')
+        cds.pop('pseudo-candidate')
+    log.info('found: pseudogenes=%i', len(pseudogenes))
+    return pseudogenes
+
+
+def get_elongated_cds(cds: dict, contig: dict, offset: int = bc.PSEUDOGENE_OFFSET) -> Tuple[dict, bool, int, int]:
+    """
+    Elongate the given CDS sequence with the offset in upstream and downstream direction, if possible.
+    """
+    tmp_cds: dict = cds.copy()
+    edge: bool = False
+    left_fill: int = 0
+    right_fill: int = 0
+
+    if contig['topology'] == 'circular' and tmp_cds['start'] - offset < 0:
+        tmp_cds['start'] = len(contig['sequence']) - (offset - tmp_cds['start'])
+        tmp_cds['edge'] = True
+        edge = True
+    elif tmp_cds['start'] - offset < 0:
+        tmp_cds['start'] = 0
+        left_fill = (tmp_cds['start'] - offset) * -1
+    else:
+        tmp_cds['start'] = tmp_cds['start'] - offset
+
+    if contig['topology'] == 'circular' and tmp_cds['stop'] + offset > len(contig['sequence']):
+        tmp_cds['stop'] = (tmp_cds['stop'] + offset) - len(contig['sequence'])
+        tmp_cds['edge'] = True
+        edge = True
+    elif tmp_cds['stop'] + offset > len(contig['sequence']):
+        tmp_cds['stop'] = len(contig['sequence'])
+        right_fill = tmp_cds['stop'] + offset - len(contig['sequence'])
+    else:
+        tmp_cds['stop'] = tmp_cds['stop'] + offset
+
+    return tmp_cds, edge, left_fill, right_fill
```

Comment:
It might be best to write a test with a few test cases to pin point the behavior.